modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/PainlessParser.java

```diff
@@ -2834,8 +2837,21 @@ public final LambdaContext lambda() throws RecognitionException {
       }
       setState(448);
       match(ARROW);
-      setState(449);
-      block();
+      setState(451);
+      switch ( getInterpreter().adaptivePredict(_input,38,_ctx) ) {
```

Comment:
All `adaptivePredict` calls are there because of *some* ambiguity. In this case I don't expect it to be particularly slow because it has to be after a lambda rather than every expression like `;` was.

It is sort of hard review the code and see the ambiguity without tracing inside of `adaptivePredict` and having a look at the path it takes here. And the path it takes comes from the big string blob.